plugins/modules/nd_federation_member.py

```diff
@@ -0,0 +1,280 @@
+
+    federation_path = "/nexus/api/federation/v4/federations"
+    member_path = "/nexus/api/federation/v4/members"
+
+    # GET local cluster name
+    local_cluster_name = nd.query_obj("/nexus/infra/api/platform/v1/clusters").get("items")[0].get("spec").get("name")
+
+    # GET federation
+    federation_obj = nd.query_obj(federation_path, ignore_not_found_error=True).get("items")
+
+    # If federation exists, verify if local_cluster is the primary
+    if federation_obj:
+        federation_info = next((federation_dict for federation_dict in federation_obj if federation_dict.get("spec").get("name") == local_cluster_name), None)
+        if not federation_info and state != "query":
+            nd.fail_json(msg="Local cluster is not the primary cluster in the federation. Cannot add/remove a member to this federation.")
+
+    # Get federation members
+    federation_member_obj = nd.query_obj(member_path, ignore_not_found_error=True).get("items")
+
+    # Query specific member
+    if clusters and state == "query":
+        if federation_member_obj:
+            for cluster in clusters:
+                cluster_info = next(
+                    (cluster_dict for cluster_dict in federation_member_obj if cluster_dict.get("spec").get("host") == cluster.get("cluster_hostname")), None
+                )
+                if cluster_info:
+                    nd.existing = cluster_info
+    else:
+        nd.existing = federation_member_obj
+
+    nd.previous = nd.existing
+
+    if state == "query":
+        nd.exit_json()
+    elif state == "absent":
+        if nd.existing:
+            if not module.check_mode:
+                for member_host in federation_member_obj:
+                    if not member_host.get("spec").get("host") == local_cluster_name:
+                        cluster_member_path = "{0}/{1}".format(member_path, member_host.get("status").get("memberID"))
+                        nd.request(cluster_member_path, method="DELETE")
+
+                # Remove the federation if there are no more members.
+                if len(nd.query_obj(member_path, ignore_not_found_error=True).get("items")) == 1:
+                    federation_info = next(
+                        (federation_dict for federation_dict in federation_obj if federation_dict.get("spec").get("name") == local_cluster_name), None
+                    )
+                    if federation_info:
+                        nd.request("{0}/{1}".format(federation_path, federation_info.get("status").get("federationID")), method="DELETE")
+            nd.existing = {}
+    elif state == "present":
+        remove_member_list = []
+        add_member_list = []
+        payload_dict = {}
+        if len(federation_member_obj) <= 1:
+            # if there are no members or just a local member, add all members from users.
+            add_member_list = clusters
+        else:
+            # Remove exisiting members not specified by the users.
+            member_exists = False
```

Comment:
Do we need this boolean flag or could we use next() function like you do above in line 197? same applies for the add equivalent in line 223

Reply:
yes, and modified